### PR TITLE
Create metric cards for mean IOU

### PR DESCRIPTION
**Pull request by sashavor**
Proposing a metric card for mIoU :rocket:

sorry for spamming you with review requests, @albertvillanova ! :hugs:

### CODE REVIEW

File: metrics/mean_iou/README.md [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+## Metric Description
+
+IoU (Intersection over Union) is the area of overlap between the predicted segmentation and the ground truth divided by the area of union between the predicted segmentation and the ground truth. 
+
+For binary (two classes) or multi-class segmentation, the *mean IoU* of the image is calculated by taking the IoU of each class and averaging them.
+
+## How to Use
+
+The Mean IoU metric takes two arrays as input corresponding to the predicted and ground truth segmentations:
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:12]:
```suggestion
The Mean IoU metric takes two numeric arrays as input corresponding to the predicted and ground truth segmentations:
```

File: metrics/mean_iou/README.md [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+>>> ground_truth = np.array([[1, 2, 2], [8, 2, 1], [3, 255, 1]])
+>>> results = mean_iou.compute(predictions=predicted, references=ground_truth, num_labels=10, ignore_index=255)
+```
+
+### Inputs
+**Mandatory inputs**
+- **predictions** (`List[ndarray]`): List of predicted segmentation maps, each of shape (height, width). Each segmentation map can be of a different size.
+- **references** (`List[ndarray]`): List of ground truth segmentation maps, each of shape (height, width). Each segmentation map can be of a different size.
+- **num_labels** (`int`): Number of classes (categories). 
+- **ignore_index** (`int`): Index that will be ignored during evaluation.
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:26]: For consistency with other metrics:
```suggestion
- `predictions` (`List[ndarray]`): List of predicted segmentation maps, each of shape (height, width). Each segmentation map can be of a different size.
- `references` (`List[ndarray]`): List of ground truth segmentation maps, each of shape (height, width). Each segmentation map can be of a different size.
- `num_labels` (`int`): Number of classes (categories). 
- `ignore_index` (`int`): Index that will be ignored during evaluation.
```

File: metrics/mean_iou/README.md [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+
+**Optional inputs**
+- **nan_to_num** (`int`): If specified, NaN values will be replaced by the number defined by the user.
+- **label_map** (`dict`): If specified, dictionary mapping old label indices to new label indices.
+- **reduce_labels** (`bool`): Whether or not to reduce all label values of segmentation maps by 1. Usually used for datasets where 0 is used for background, and background itself is not included in all classes of a dataset (e.g. ADE20k). The background label will be replaced by 255. The default value is `False`.
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:31]:
```suggestion
- `nan_to_num` (`int`): If specified, NaN values will be replaced by the number defined by the user.
- `label_map` (`dict`): If specified, dictionary mapping old label indices to new label indices.
- `reduce_labels` (`bool`): Whether or not to reduce all label values of segmentation maps by 1. Usually used for datasets where 0 is used for background, and background itself is not included in all classes of a dataset (e.g. ADE20k). The background label will be replaced by 255. The default value is `False`.
```

File: metrics/mean_iou/README.md [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+### Output Values
+The metric returns a dictionary with the following elements:
+- **mean_iou** (`float`): Mean Intersection-over-Union (IoU averaged over all categories).
+- **mean_accuracy** (`float`): Mean accuracy (averaged over all categories). 
+- **overall_accuracy** (`float`): Overall accuracy on all images.
+- **per_category_accuracy** (`ndarray` of shape `(num_labels,)`): Per category accuracy.
+- **per_category_iou** (`ndarray` of shape `(num_labels,)`): Per category IoU.
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:39]:
```suggestion
- `mean_iou` (`float`): Mean Intersection-over-Union (IoU averaged over all categories).
- `mean_accuracy` (`float`): Mean accuracy (averaged over all categories). 
- `overall_accuracy` (`float`): Overall accuracy on all images.
- `per_category_accuracy` (`ndarray` of shape `(num_labels,)`): Per category accuracy.
- `per_category_iou` (`ndarray` of shape `(num_labels,)`): Per category IoU.
```

File: metrics/mean_iou/README.md [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+license = {Apache-2.0},
+month = {7},
+title = {{OpenMMLab Semantic Segmentation Toolbox and Benchmark}},
+url = {https://github.com/open-mmlab/mmsegmentation},
+year = {2020}
+}"
+```
+
+
+## Further References
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:90]: No further references?

File: metrics/mean_iou/README.md [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+>>> # suppose one has 3 different segmentation maps predicted
+>>> predicted_1 = np.array([[1, 2], [3, 4], [5, 255]])
+>>> actual_1 = np.array([[0, 3], [5, 4], [6, 255]])
+>>> predicted_2 = np.array([[2, 7], [9, 2], [3, 6]])
+>>> actual_2 = np.array([[1, 7], [9, 2], [3, 6]])
+>>> predicted_3 = np.array([[2, 2, 3], [8, 2, 4], [3, 255, 2]])
+>>> actual_3 = np.array([[1, 2, 2], [8, 2, 1], [3, 255, 1]])
+>>> predicted = [predicted_1, predicted_2, predicted_3]
+>>> ground_truth = [actual_1, actual_2, actual_3]
+>>> results = mean_iou.compute(predictions=predicted, references=ground_truth, num_labels=10, ignore_index=255, reduce_labels=False)
```
Comment (albertvillanova) [inline metrics/mean_iou/README.md:68]: Nit, feel free to ignore it.
```suggestion
>>> predictions = [predicted_1, predicted_2, predicted_3]
>>> references = [actual_1, actual_2, actual_3]
>>> results = mean_iou.compute(predictions=predictions, references=references, num_labels=10, ignore_index=255, reduce_labels=False)
```

File: metrics/mean_iou/README.md (hunk 5 above)
Comment (albertvillanova) [reply metrics/mean_iou/README.md:90]: @sashavor not clear candidate... Maybe the Wikipedia article?
https://en.wikipedia.org/wiki/Jaccard_index

Comment (sashavor) [reply metrics/mean_iou/README.md:90]: great idea!

**Review by albertvillanova — APPROVED** [review]
Thanks!

**Pull request merged**